src/main.ts

```diff
@@ -299,6 +313,84 @@ class CNLambdaFunctionAspect implements IAspect {
 }
 Aspects.of(app).add(new CNLambdaFunctionAspect());
 
+class RoleNamePrefixAspect implements IAspect {
+  private prefix: string;
+
+  constructor(prefix: string) {
+    this.prefix = prefix;
+  }
+
+  public visit(node: IConstruct): void {
+    if (Role.isRole(node) && node.stack.toString().startsWith('private-exist-vpc-')) {
+      let roleId = node.node.id;
+      if (roleId === 'ServiceRole' || roleId === 'Role') {
+        const parentId = node.node.scope?.node.id;
+        roleId = `${parentId}${roleId}`;
+      }
+      const cfnRole = node.node.defaultChild as CfnRole;
+      cfnRole.addPropertyOverride('RoleName', this.getRoleName(this.prefix, roleId, getShortIdOfStackWithRegion(Stack.of(node))));
+      addCfnNagSuppressRules(
+        cfnRole,
+        [
+          {
+            id: 'W28',
+            reason: 'Set the role name with prefix',
+          },
+        ],
+      );
+    } else if (CfnResource.isCfnResource(node) && node.cfnResourceType === 'AWS::IAM::Role') {
+      if (!(node as CfnRole).roleName && node.stack.toString().startsWith('private-exist-vpc-')) {
+        const parentId = node.node.scope?.node.id;
+        node.addPropertyOverride('RoleName', this.getRoleName(this.prefix, `${parentId}Role`, getShortIdOfStackWithRegion(Stack.of(node))));
+        addCfnNagSuppressRules(
+          node,
+          [
+            {
+              id: 'W28',
+              reason: 'Set the role name with prefix',
+            },
+          ],
+        );
+      }
+    }
+  }
+
+  private getRoleName(prefix: string, id: string, suffix: string): string {
+    const roleNameMaxLength = 64;
+    const splitLength = 2;
+    const prefixLength = prefix.length;
+    const suffixLength = 20;
+    if (prefixLength + suffixLength + splitLength > roleNameMaxLength) {
+      throw new Error(`Role name length is greater than ${roleNameMaxLength}`);
+    }
+    const nameLength = roleNameMaxLength - prefixLength - suffixLength - splitLength;
+    if (id.length < nameLength) {
+      return `${prefix}-${id}-${suffix}`;
+    }
+    const name = id.substring(id.length - nameLength, id.length);
+    return `${prefix}-${name}-${suffix}`;
+  };
+}
+if (iamRolePrefix) {
+  Aspects.of(app).add(new RoleNamePrefixAspect(iamRolePrefix));
+}
+
+class RolePermissionBoundaryAspect implements IAspect {
+  readonly permissionBoundary: string;
+  constructor(permissionBoundary: string) {
+    this.permissionBoundary = permissionBoundary;
+  }
+
+  public visit(node: IConstruct): void {
+    if (CfnResource.isCfnResource(node) && node.cfnResourceType === 'AWS::IAM::Role') {
+      node.addPropertyOverride('PermissionsBoundary', `arn:${Aws.PARTITION}:iam::${Aws.ACCOUNT_ID}:policy/${this.permissionBoundary}`);
+    }
+  }
+}
+if (app.node.tryGetContext('rolePermissionBoundary')) {
+  Aspects.of(app).add(new RolePermissionBoundaryAspect(app.node.tryGetContext('rolePermissionBoundary')));
```

Comment:
move to use CfnParameter instead of CDK context, which only works on deploying from source code